espnet2/asr/espnet_model.py

```diff
@@ -79,20 +85,39 @@ def __init__(
             self.ctc = None
         else:
             self.ctc = ctc
-        self.rnnt_decoder = rnnt_decoder
+
+        self.transducer_decoder = transducer_decoder
+        self.joint_network = joint_network
+
+        self.criterion_transducer = TransLoss("warp-transducer", self.blank)
+
         self.criterion_att = LabelSmoothingLoss(
             size=vocab_size,
             padding_idx=ignore_id,
             smoothing=lsm_weight,
             normalize_length=length_normalized_loss,
         )
 
+        self.error_calculator = None
+        self.error_calculator_trans = None
         if report_cer or report_wer:
-            self.error_calculator = ErrorCalculator(
-                token_list, sym_space, sym_blank, report_cer, report_wer
-            )
-        else:
-            self.error_calculator = None
+            if transducer_decoder:
+                self.error_calculator_trans = ErrorCalculatorTransducer(
+                    self.transducer_decoder,
+                    self.joint_network,
+                    token_list,
+                    sym_space,
+                    sym_blank,
+                    report_cer,
+                    report_wer,
+                )
+
+            if not transducer_decoder or (
+                transducer_decoder and transducer_weight != 0.0
```

Comment:
I guess the second condition should be `transducer_decoder and transducer_weight == 0.0`.

Reply:
> I guess the second condition should be transducer_decoder and transducer_weight == 0.0.

I think the second condition is correct?

CER/WER computation for other models should only be enabled if 1) no transducer decoder is selected and 2) if a transducer decoder is selected and multi-task learning with transducer is enabled (`transducer_weight != 0.0`)

The difference between ESPnet1 and ESPnet2 is that transducer loss can be used in multi-task learning (current form is temporary though.)

Reply:
Oh I see. I misunderstood the multi-task part.